app/lib/features/space/sheets/link_room_sheet.dart

```diff
@@ -0,0 +1,422 @@
+        childRoomsIds.add(space.knownSubspaces[i].getRoomId().toString());
+      }
+      //Add recommended child spaces ids
+      recommendedChildSpaceIds.clear();
+      for (int i = 0; i < space.otherRelations.length; i++) {
+        recommendedChildSpaceIds
+            .add(space.otherRelations[i].getRoomId().toString());
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+
+    return SideSheet(
+      header: sheetHeader,
+      body: SizedBox(
+        height: size.height - 120,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            parentSpaceDataUI(),
+            searchUI(),
+            Expanded(child: roomList()),
+          ],
+        ),
+      ),
+    );
+  }
+
+  //Search
+  Widget searchUI() {
+    return Search(
+      onChanged: (value) {
+        if (childRoomType == ChildRoomType.chat) {
+          ref.read(chatSearchValueProvider.notifier).update((state) => value);
+        } else if (childRoomType == ChildRoomType.space) {
+          ref.read(spaceSearchValueProvider.notifier).update((state) => value);
+        }
+      },
+      searchController: searchTextEditingController,
+    );
+  }
+
+  //Parent space
+  Widget parentSpaceDataUI() {
+    final spaceDetails = ref.watch(selectedSpaceDetailsProvider);
+    return Padding(
+      padding: const EdgeInsets.all(12),
+      child: spaceDetails.when(
+        data: (space) {
+          return space == null
+              ? const SizedBox.shrink()
+              : Column(
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    const Text('Parent Space'),
+                    SpaceChip(space: space),
+                  ],
+                );
+        },
+        error: (e, s) => errorUI('error: ', e),
+        loading: () => Container(),
+      ),
+    );
+  }
+
+  //Manage list data based on the child room type
+  Widget roomList() {
+    if (childRoomType == ChildRoomType.chat) {
+      return chatsList();
+    } else if (childRoomType == ChildRoomType.space ||
+        childRoomType == ChildRoomType.recommendedSpace) {
+      return spacesList();
+    } else {
+      return const SizedBox.shrink();
+    }
+  }
+
+  //List of chats excluding DMs that can be linked according to the selected parent space
+  Widget chatsList() {
+    final searchValue = ref.watch(chatSearchValueProvider);
+    if (searchValue != null && searchValue.isNotEmpty) {
+      return searchedChatsList();
+    }
+
+    final chatList =
+        ref.watch(chatsProvider).where((element) => (!element.isDm())).toList();
+    return chatListUI(chatList);
+  }
+
+  //Show chat list based on the search term
+  Widget searchedChatsList() {
+    return ref.watch(searchedChatsProvider).when(
```

Comment:
I know @gtalha07 had some feedback on having function that return `Widget`s, especially if they use `ref.watch` internally... Maybe you want to share...

Reply:
@gnunicorn 
Not getting your feedback, can you elaborate more about this feedback?

Reply:
I'd leave this to @gtalha07 ....

Reply:
> I know @gtalha07 had some feedback on having function that return `Widget`s, especially if they use `ref.watch` internally... Maybe you want to share...

Could you confirm whether this works without any issues? The ```ref.watch()``` inside widget ```chatList()```?

Reply:
I believe that `ref.watch()` will work inside widget `chatList()` without any issue.

The main reason is that it is used in the `ConsumerStatefulWidget` class and based on the riverpod core concept we can use `ref.watch()` anywhere in class which extends `ConsumerStatefulWidget`.

`ref` is define as a state in the riverpod `ConsumerStatefulWidget` class. 

Reference:
<img width="1036" alt="Screenshot 2023-11-02 at 9 20 44 PM" src="https://github.com/acterglobal/a3/assets/51526480/d4c69e32-c031-40b3-b37a-eb8c91b5ad05">

@gnunicorn 
Still share your thoughts here if I missed or miss understand anything about riverpod concept.

Reply:
@gtalha07 
Still I’m unsure about the `ref.watch()` with one time listening or continues listening in a function.

@gnunicorn please share your thoughts here if have any idea about this concept

Reply:
I haven't faced any issue on feature implementation.